test/lit/passes/outlining.wast

```diff
@@ -49,60 +59,152 @@
 )
 
 ;; Tests that outlining occurs properly when the sequence is at the end of a function.
-
-;; CHECK:      (func $outline$ (param $0 i32)
-;; CHECK-NEXT:  (drop
-;; CHECK-NEXT:   (local.get $0)
-;; CHECK-NEXT:  )
-;; CHECK-NEXT:  (drop
-;; CHECK-NEXT:   (i32.const 1)
-;; CHECK-NEXT:  )
-;; CHECK-NEXT:  (drop
-;; CHECK-NEXT:   (i32.const 2)
-;; CHECK-NEXT:  )
-;; CHECK-NEXT: )
 (module
   ;; CHECK:      (type $0 (func))
 
   ;; CHECK:      (type $1 (func (param i32)))
 
-  ;; CHECK:      (func $a
+  ;; CHECK:      (func $outline$ (param $0 i32)
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (local.get $0)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.const 2)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+
+  ;; CHECK:      (func $c
   ;; CHECK-NEXT:  (drop
   ;; CHECK-NEXT:   (i32.const 7)
   ;; CHECK-NEXT:  )
   ;; CHECK-NEXT:  (call $outline$
   ;; CHECK-NEXT:   (i32.const 4)
   ;; CHECK-NEXT:  )
   ;; CHECK-NEXT: )
-  (func $a
+  (func $c
     (drop (i32.const 7))
     (drop (i32.const 4))
     (drop (i32.const 1))
     (drop (i32.const 2))
   )
-  ;; CHECK:      (func $b
+  ;; CHECK:      (func $d
   ;; CHECK-NEXT:  (drop
   ;; CHECK-NEXT:   (i32.const 0)
   ;; CHECK-NEXT:  )
   ;; CHECK-NEXT:  (call $outline$
   ;; CHECK-NEXT:   (i32.const 5)
   ;; CHECK-NEXT:  )
   ;; CHECK-NEXT: )
-  (func $b
+  (func $d
     (drop (i32.const 0))
     (drop (i32.const 5))
     (drop (i32.const 1))
     (drop (i32.const 2))
   )
 )
-;; CHECK:      (func $outline$ (param $0 i32)
-;; CHECK-NEXT:  (drop
-;; CHECK-NEXT:   (local.get $0)
-;; CHECK-NEXT:  )
+
+;; Tests that outlining occurs properly when the sequence is at the beginning of a function.
+;; Also tests that the outlined function has no arguments.
+(module
+  ;; CHECK:      (type $0 (func))
+
+  ;; CHECK:      (func $outline$
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.const 0)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.add
+  ;; CHECK-NEXT:    (i32.const 0)
+  ;; CHECK-NEXT:    (i32.const 1)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
```

Comment:
I think we could simplify the outlined sequence here.

Reply:
done, also simplified the second test in the file